src/commands/workspaces.ts

```diff
@@ -0,0 +1,172 @@
+			};
+		}));
+
+	const picked = await vscode.window.showQuickPick(pickItemsPromise, { canPickMany: false, placeHolder });
+	return picked;
+}
+
+export class ConnectInNewWindowCommand implements Command {
+	readonly id = 'gitpod.workspaces.connectInNewWindow';
+
+	constructor(
+		private readonly context: vscode.ExtensionContext,
+		private readonly sessionService: ISessionService,
+		private readonly hostService: IHostService,
+	) { }
+
+	async execute(treeItem?: { id: string }) {
+		let wsData: WorkspaceData | undefined;
+		if (!treeItem?.id) {
+			wsData = await showWorkspacesPicker(this.sessionService, 'Select a workspace to connect...');
+		} else {
+			wsData = rawWorkspaceToWorkspaceData(await this.sessionService.getAPI().getWorkspace(treeItem.id));
+		}
+
+		if (!wsData) {
+			return;
+		}
+
+		const domain = getLocalSSHDomain(this.hostService.gitpodHost);
+		const sshHostname = `${wsData.id}.${domain}`;
+		const sshDest = new SSHDestination(sshHostname, wsData.id);
+
+		// TODO: remove this, should not be needed
+		await this.context.globalState.update(`${SSH_DEST_KEY}${sshDest.toRemoteSSHString()}`, { workspaceId: wsData.id, gitpodHost: this.hostService.gitpodHost, instanceId: '' } as SSHConnectionParams);
+
+		// TODO: getWorkspace API need to return path to open, for now harcode it
+		await vscode.commands.executeCommand(
+			'vscode.openFolder',
+			vscode.Uri.parse(`vscode-remote://ssh-remote+${sshDest.toRemoteSSHString()}/workspace/${wsData.repo}`),
+			{ forceNewWindow: true }
+		);
+	}
+}
+
+export class ConnectInCurrentWindowCommand implements Command {
+	readonly id = 'gitpod.workspaces.connectInCurrentWindow';
+
+	constructor(
+		private readonly context: vscode.ExtensionContext,
+		private readonly sessionService: ISessionService,
+		private readonly hostService: IHostService,
+	) { }
+
+	async execute(treeItem?: { id: string }) {
+		let wsData: WorkspaceData | undefined;
+		if (!treeItem?.id) {
+			wsData = await showWorkspacesPicker(this.sessionService, 'Select a workspace to connect...');
+		} else {
+			wsData = rawWorkspaceToWorkspaceData(await this.sessionService.getAPI().getWorkspace(treeItem.id));
+		}
+
+		if (!wsData) {
+			return;
+		}
+
+		const domain = getLocalSSHDomain(this.hostService.gitpodHost);
+		const sshHostname = `${wsData.id}.${domain}`;
+		const sshDest = new SSHDestination(sshHostname, wsData.id);
+
+		// TODO: remove this, should not be needed
+		await this.context.globalState.update(`${SSH_DEST_KEY}${sshDest.toRemoteSSHString()}`, { workspaceId: wsData.id, gitpodHost: this.hostService.gitpodHost, instanceId: '' } as SSHConnectionParams);
+
+		// TODO: getWorkspace API need to return path to open, for now harcode it
+		await vscode.commands.executeCommand(
+			'vscode.openFolder',
+			vscode.Uri.parse(`vscode-remote://ssh-remote+${sshDest.toRemoteSSHString()}/workspace/${wsData.repo}`),
+			{ forceNewWindow: false }
+		);
+	}
+}
+
+export class StopEnvironmentCommand implements Command {
```

Comment:
Can we don't use Envronment. It is something which we don't have in other code, let's stick to usual terminology, i.e. `StopWorkspace` or `StopWorkspaceInstance`?